yt/yt/server/query_tracker/profiler.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <yt/yt/library/profiling/sensor.h>
+
+#include <yt/yt/client/query_tracker_client/public.h>
+
+#include <library/cpp/yt/threading/spin_lock.h>
+#include <library/cpp/yt/threading/public.h>
+
+namespace NYT::NQueryTracker {
+
+const TString NoneQueryTracker = "None";
+
+////////////////////////////////////////////////////////////////////////////////
+
+struct TProfilingTags
+{
+    EQueryState State;
+    EQueryEngine Engine;
+    TString AssignedTracker;
+
+    bool operator==(const TProfilingTags& other) const = default;
+    bool operator<(const TProfilingTags& other) const = default;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+
+} // namespace NYT::NQueryTracker
+
+////////////////////////////////////////////////////////////////////////////////
+
+template <>
```

Comment:
usually we don't break the internals of a namespace, and declare THash in the end of the file